FlatBuffer.hs: Replicate Rust module name generation logic of flatc.

ryzhyk opened the pull request:
We used the `Casing.quietSnake` function to generate snake-case Rust
module name when importing the `flatc`-generated module.  This function
does not precisely match the algorithm used by `flatc`, causing "unresolved
import" errors.

Instead of using the library function, we precisely re-implement the
algorithm used by `flatc` in Haskell.

Resolves #425

d-e-s-o commented:
Thanks for the fix! I second Mihai's comment. I guess at the least we could rename one of the existing tests to match the pattern (as that's also how I triggered it). Have you checked whether we can remove `casing` from `package.yml` now?

ryzhyk commented:
> Thanks for the fix! I second Mihai's comment. I guess at the least we could rename one of the existing tests to match the pattern (as that's also how I triggered it). Have you checked whether we can remove `casing` from `package.yml` now?

Added a test.  And removed `casing`.  Thanks for spotting!

mbudiu-vmw commented on test/datalog_tests/dcm1.dl:
```diff
@@ -0,0 +1,77 @@
+import intern
```
no reference file?

ryzhyk replied:
Test data coming in a few days..
